ee/ucp/kubernetes/layer-7-routing.md

```diff
@@ -19,87 +21,180 @@ Use an ingress controller when you want to:
+dockeree-worker-linux-2    Ready     <none>    5d        v1.8.11-docker-8d637ae
+dockeree-worker-linux-3    Ready     <none>    5d        v1.8.11-docker-8d637ae
 
-## Create a grant
+```
 
-The default service account that's associated with the `ingress-nginx`
-namespace needs access to Kubernetes resources, so create a grant with
-`Restricted Control` permissions.
+The label `infra.role=ingress` is used to label the nodes:
 
-1.  From UCP, navigate to the **Grants** page, and click **Create Grant**.
-2.  Within the **Subject** pane, select **Service Account**. For the
-    **Namespace** select **ingress-nginx**, and select **default** for
-    the **Service Account**. Click **Next**.
-3.  Within the **Role** pane, select **Restricted Control**, and then click
-    **Next**.
-4.  Within the **Resource Set** pane, select the **Type** **Namespace**, and
-    select the **Apply grant to all existing and new namespaces** toggle.
-5.  Click **Create**.
+```
+🐳  → kubectl label node dockeree-worker-linux-1 infra.role=ingress
+node "dockeree-worker-linux-1" labeled
+🐳  → kubectl label node dockeree-worker-linux-2 infra.role=ingress
+node "dockeree-worker-linux-2" labeled
+🐳  → kubectl label node dockeree-worker-linux-3 infra.role=ingress
+node "dockeree-worker-linux-3" labeled
 
-## Deploy NGINX ingress controller
+```
 
-The cluster is ready for the ingress controller deployment, which has three
-main components:
+### Step 2: Create a Dedicated Namespace
 
-- a simple HTTP server, named `default-http-backend`,
-- an ingress controller, named `nginx-ingress-controller`, and
-- a service that exposes the app, named `ingress-nginx`.
+A dedicated namespace, for example, `nginx-controller`, is needed for all infrastructure deployment activities. You also need a service account to enable the ingress controller to work with the Kubernetes API. After creating a namespace and a service account, you must create an RBAC policy to only allow infrastructure deployments on the dedicated nodes that were labelled in the previous example.
 
-Navigate to the **Create Kubernetes Object** page, and in the **Object YAML**
-editor, paste the following YAML.
+To create a namespace and a service account, simply use the following [YAML file](config/ns-and-sa.yaml) and apply it via the CLI or the UI:
 
-```yaml
-apiVersion: apps/v1beta2
+```
+🐳  → cat ns-and-sa.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: infra
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: nginx-ingress-service-account
+  namespace: infra
+```
+
+Then apply the namespace and service account via `kubectl`:
+
+```
+🐳  → kubectl apply -f ns-and-sa.yaml
+namespace "infra" created
+serviceaccount "nginx-ingress-service-account" created
+
+```
+
+### Step 3: Create an RBAC Policy
+
+Apply an RBAC role-binding for NGINX controller access to the API Server, as shown in the following [RBAC yaml config](config/ingress-rbac.yaml) example:
```

Comment:
please be aware that i reference the configs that were in my repo ( under the `config`) sub directory. No need to reference configs if you're documenting them directly in this doc.

Reply:
Thanks - removing reference.